### PR TITLE
Merge master into v2

### PR DESCRIPTION
**Pull request by padmaia**

### CODE REVIEW

File: packages/core/parcel-bundler/src/scope-hoisting/concat.js [hunk 1]
```diff
@@ -161,6 +163,29 @@ module.exports = (packager, ast) => {
             if (!isUnusedValue(path)) {
               let name = getName(mod, 'exports');
               node = t.identifier(replacements.get(name) || name);
+
+              // Insert __esModule interop flag if the required module is an ES6 module with a default export.
+              // This ensures that code generated by Babel and other tools works properly.
+              if (
+                asset.cacheData.isCommonJS &&
```
Comment (wbinnssmith) [inline packages/core/parcel-bundler/src/scope-hoisting/concat.js]: @mischnic was this ported to v2?

File: packages/core/parcel-bundler/src/scope-hoisting/hoist.js [hunk 2]
```diff
@@ -259,13 +268,23 @@ module.exports = {
...
-        let [decl] = path.insertBefore(
-          t.variableDeclaration('var', [
-            t.variableDeclarator(t.clone(identifier), right)
-          ])
-        );
 
-        scope.registerDeclaration(decl);
+        // If in the program scope, create a variable declaration and initialize with the exported value.
+        // Otherwise, declare the variable in the program scope, and assign to it here.
+        if (path.scope === scope) {
```
Comment (wbinnssmith) [inline packages/core/parcel-bundler/src/scope-hoisting/hoist.js]: @mischnic same with these 😃

Comment (mischnic) [reply packages/core/parcel-bundler/src/scope-hoisting/hoist.js]: Yes: https://github.com/parcel-bundler/parcel/blob/cb18d4507647f31824daeab1dc9de9428321a02a/packages/shared/scope-hoisting/src/hoist.js#L271-L286

File: packages/core/parcel-bundler/src/scope-hoisting/concat.js (hunk 1 above)
Comment (mischnic) [reply packages/core/parcel-bundler/src/scope-hoisting/concat.js]: Yes (actually, Devon ported these to Parcel 1): https://github.com/parcel-bundler/parcel/blob/cb18d4507647f31824daeab1dc9de9428321a02a/packages/shared/scope-hoisting/src/link.js#L190-L215

**Pull request merged**